### PR TITLE
Correctly handle non-ascii header values

### PR DESCRIPTION
**Pull request by jamesls**
Fixes aws/aws-cli#1124

cc @kyleknap @danielgtaylor

### CODE REVIEW

File: botocore/awsrequest.py [hunk 1]
```diff
@@ -123,7 +123,9 @@ def _send_request(self, method, url, body, headers):
 
     def _send_output(self, message_body=None):
         self._buffer.extend((b"", b""))
-        msg = b"\r\n".join(self._buffer)
+        msg = b"\r\n".join(
+            b.encode('utf-8') if isinstance(b, six.text_type) else b
```
Comment (kyleknap) [inline botocore/awsrequest.py]: I'd prefer if it was not all in-line as it took me a little longer to understand. Code looks fine otherwise.

**Comment by kyleknap** [comment]
Looks good. :ship: Had small comment.

File: botocore/awsrequest.py (hunk 1 above)
Comment (jamesls) [inline botocore/awsrequest.py]: Ok, extracted that out into it's own method.

Comment (kyleknap) [inline botocore/awsrequest.py]: Thanks. Looks good.

**Pull request merged**